### PR TITLE
32-bit support for xxHash algorithm

### LINKED ISSUE #1: Support for 32-bit xxHash algorithm
See associated PR: #2

### PR DESCRIPTION
**Pull request by ghost**
Closes #1

### CODE REVIEW

**Comment by ekpyron** [comment]
Can you perhaps add a handful of canonical test cases to ``test.cpp``? Then I'd be happy to merge this, thank you very much!

File: xxh32.hpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+*/
+
+#ifndef XXHASHCT_XXH32_HPP
+#define XXHASHCT_XXH32_HPP
+
+#include <cstdint>
+
+struct xxh32 {
+	static constexpr uint32_t hash( const char *p, uint32_t len, uint32_t seed ) {
+		return finalize( (len >= 16 ? h32bytes(p, len, seed) : seed+P5) + len, p+(len & ~0xF), len & 0xF );
```
Comment (ekpyron) [inline xxh32.hpp]:
```suggestion
		return finalize( (len >= 16 ? h16bytes(p, len, seed) : seed+P5) + len, p+(len & ~0xF), len & 0xF );
```

**Comment by ekpyron** [comment]
As a matter of fact, I just tried to extend the tests in ``test.cpp`` to your 32-bit implementation and checked against the reference implementation https://github.com/Cyan4973/xxHash/, but not all tests succeed, so there must be some small incorrectness in the implementation - I'll have a more detailed look later.

**Comment by ghost** [comment]
To be honest, it's been a long time since I wrote this and as far as I can remember, I didn't came into any problems when I used it. But anyway, I'll look at it soon and make any edits needed.

**Comment by ekpyron** [comment]
I'm just noticing that shockingly this has been open for a year now... I'll try looking into it soon, but I can't promise when I'll get around to.

**Comment by ghost** [comment]
Sorry for the force-push mess above, I initially committed with an old account name and wanted to get rid of it.

I forgot about this pull request but finally took the necessary time to finish it.
I added the tests to `test.cpp` as you requested and corrected my implementation to make them pass.

**Review by ekpyron — APPROVED** [review]
Nice! Thank you very much!

**Pull request merged**